test/spectests/spectests.cpp

```diff
@@ -404,16 +422,22 @@ class test_runner
 
     std::optional<fizzy::Value> read_value(const json& v)
     {
-        const auto arg_type = v.at("type").get<std::string>();
-        if (arg_type != "i32" && arg_type != "i64" && arg_type != "f32" && arg_type != "f64")
+        const auto type = v.at("type").get<std::string>();
+
+        // JSON tests have all values including floats serialized as 64-bit unsigned integers
+        const uint64_t uint_value = std::stoull(v.at("value").get<std::string>());
+
+        if (type == "i32" || type == "i64")
+            return uint_value;
```

Comment:
I decided for now not to complicate it with checking if the value declared as 32-bit is really not wider. It won't result in UB if it happens.